applications/ChimeraApplication/custom_strategies/strategies/fs_strategy_for_chimera.h

```diff
@@ -704,6 +706,18 @@ class FractionalStepStrategyForChimera : public FractionalStepStrategy<TSparseSp
     }
 
 
+    double SolveStrategy(StrategyPointerType& pStrategy)
+    {
+        pStrategy->Initialize();
+        pStrategy->InitializeSolutionStep();
+        pStrategy->Predict();
+        pStrategy->SolveSolutionStep();
+        const double norm_dx = TSparseSpace::TwoNorm(pStrategy->GetSolutionVector());
+        pStrategy->FinalizeSolutionStep();        
+        return norm_dx;
+    }
```

Comment:
this is exactly what one should NOT do
=> those calls come from the AnalysisStage and shouldn't be done internally

Reply:
For fractional step strategy, this is not possible as there are two (momentum and pressure) internal strategies which should be called at different places. That is why it is done like this.

Reply:
Check how and where this function is called and used.

Reply:
Then I guess the FractionalStep Strategy hasn't been refactored yet
@rubenzorrilla can you comment?

Reply:
I don't know what @rubenzorrilla  has in works, but I believe that for fractional step solver, this is inevitable ! 
Each "Solve" of the fractional step looks something as follows 

```
SolveSoltuionStep() # Of complete fractional step solver 
    SetFractionalStep(1)
    TillConvergence :
        SolveMomentumStrategy() # HERE : SolveStrategy() is used

    ComputeOSSProjections()
    Compute/Update/Backup Pressure()
    SetFractionalStep(2)
    .... BunchOfOtherOpps()
    SolvePressureStrategy()  # HERE : SolveStrategy() is used
    UpdateEndOfStepVelocity()
```
So I am not sure if this would fit in the AnalysisStage. Also AnalysisStage does not have access to all those intermediate steps and functions !

Reply:
I think instead of `SolveStrategy` only `SolveSolutionStep` should be called

Reply:
I didn't have time to remove the `Solve` from the base `FSStrategy`. In any case, I think it would be better to first take care of that and then update the Chimera one accordingly.

Reply:
@rubenzorrilla  Thank you for the comment. If you have gone through my changes here and you say this is the way to go , I can do it in the fluid dynamics application.